Update AjaxController.php

ramandi opened the pull request:
To avoid 500 Internal Server Error while removing the "Do not contact" tag from a contact's preferences, as described in https://github.com/mautic/mautic/issues/9017


| Q                                      | A
| -------------------------------------- | ---
| Branch?                                | 3.0 
| Bug fix?                               | yes
| New feature?                           | no
| Deprecations?                          | no
| BC breaks?                             | no
| Automated tests included?              | no
| Related user documentation PR URL      | 
| Related developer documentation PR URL | 
| Issue(s) addressed                     | Fixes #9017


#### Description:
Change model name (mautic.lead.model.dnc -> lead.dnc) to avoid error.  getModel composes the full name from the short one.

ramandi committed:
Update AjaxController.php

To avoid 500 Internal Server Error while removing the "Do not contact" tag from a contact's preferences, as described in https://github.com/mautic/mautic/issues/9017

wise-gorilla reviewed (changes requested):
Please also check line #350

ramandi committed:
Update AjaxController.php

Changed large model name to short one in line 351.  getModel will reconstruct the large name.

RCheesley commented:
@cla-bot check

RCheesley commented:
Hi @ramandi thanks for making this PR! 

It turned out that the DNC issue was much deeper than we originally thought, and a PR has been created in #9186 which is attempting to pick up all the instances where DNC had become broken rather than just the single one that is covered in this PR.

I noticed that your PR has an additional line we had not captured in that PR so I've asked for it to be added.

I'd like to go ahead and close this PR down in favour of #9186 but we will ensure that you are credited in the release notes as contributing to this PR.

jacobshilitz commented:
Hello, 
1. My PR #9186 is not related to this PR.
2. `mautic.lead.model.dnc` is declared in more [than one place](https://github.com/mautic/mautic/search?q=%22mautic.lead.model.dnc%22&unscoped_q=%22mautic.lead.model.dnc%22) if it has to be replaced, then it has to be done everywhere
